addons/project/models/project.py

```diff
@@ -495,7 +495,7 @@ def map_tasks(self, new_project_id):
             new_task = task.copy(defaults)
             # If child are created before parent (ex sub_sub_tasks)
             new_child_ids = [old_to_new_tasks[child.id] for child in task.child_ids if child.id in old_to_new_tasks]
-            tasks.browse(new_child_ids).write({'parent_id': new_task.id})
+            tasks.browse(new_child_ids).write({'parent_id': new_task.id, 'display_project_id': False})
```

Comment:
I don't really agree to do that because if one of those subtasks has the , we could set the new project in the `display_project_id` field for the subtasks in which the project in the self is contained in the `display_project_id` field. 

we could perhaps do something like this: 
```diff
diff --git a/addons/project/models/project.py b/addons/project/models/project.py
index b590435eabc..a6a5163a21b 100644
--- a/addons/project/models/project.py
+++ b/addons/project/models/project.py
@@ -488,14 +488,18 @@ class Project(models.Model):
             defaults = self._map_tasks_default_valeus(task, project)
             if task.parent_id:
                 # set the parent to the duplicated task
-                defaults['parent_id'] = old_to_new_tasks.get(task.parent_id.id, False)
+                parent_id = old_to_new_tasks.get(task.parent_id.id, False)
+                defaults['parent_id'] = parent_id
+                if not parent_id:
+                    defaults['project_id'] = project.id if task.display_project_id == self else False
+                    defaults['display_project_id'] = project.id if task.display_project_id == self else False
             elif task.display_project_id == self:
                 defaults['project_id'] = project.id
                 defaults['display_project_id'] = project.id
             new_task = task.copy(defaults)
             # If child are created before parent (ex sub_sub_tasks)
             new_child_ids = [old_to_new_tasks[child.id] for child in task.child_ids if child.id in old_to_new_tasks]
-            tasks.browse(new_child_ids).write({'parent_id': new_task.id, 'display_project_id': False})
+            tasks.browse(new_child_ids).write({'parent_id': new_task.id})
             old_to_new_tasks[task.id] = new_task.id
             if task.allow_task_dependencies:
                 depend_on_ids = [t.id if t not in all_tasks else old_to_new_tasks.get(t.id, None) for t in
```

Reply:
Much better, effectively :+1: